.github/workflows/testsuite.yml

```diff
@@ -0,0 +1,406 @@
+        with:
+          fetch-depth: 1000
+      - name: git setup
+        run: |
+          echo "Pull request authors"
+          echo "# git merge-base origin/${BASE_REF} HEAD"
+          git remote set-url origin "https://$GITHUB_ACTOR:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY"
+          git config diff.renameLimit 999999
+          git fetch --depth=1000 origin ${BASE_REF}
+        env:
+          BASE_REF: ${{ github.base_ref }}
+      - name: Involved authors
+        run: |
+          git log --pretty=format:"Author: %an <%ae>" origin/${BASE_REF}...${SHA}^2 | sort -u
+        env:
+          BASE_REF: ${{ github.base_ref }}
+          SHA: ${{ github.sha }}
+
+  #  ___            _ _           ___ _           _
+  # / __| __ _ _ _ (_) |_ _  _   / __| |_  ___ __| |__
+  # \__ \/ _` | ' \| |  _| || | | (__| ' \/ -_) _| / /
+  # |___/\__,_|_||_|_|\__|\_, |  \___|_||_\___\__|_\_\
+  #                       |__/
+  # ascii art small font from https://ascii.co.uk/text
+
+  sanity_check:
+    name: "Sanity: Linux -Dusethreads"
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+
+    env:
+      PERL_SKIP_TTY_TEST: 1
+      CONTINUOUS_INTEGRATION: 1
+      WORKSPACE: ${{ github.workspace }}
+
+    steps:
+      - name: Dump GitHub context
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: echo "$GITHUB_CONTEXT"
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: git cfg + fetch tags
+        run: |
+          git config diff.renameLimit 999999
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+      - name: Configure
+        run: |
+          ./Configure -des -Dusedevel ${CONFIGURE_ARGS} -Dprefix="$HOME/perl-blead" -DDEBUGGING
+        env:
+          CONFIGURE_ARGS: "-Dusethreads"
+      - name: Build
+        run: |
+          make -j2
+      - name: Run Tests
+        run: |
+          TEST_JOBS=2 make -j2 test
+
+  # use a dedicated job to check if full testsuite is enabled
+  #   avoid starting VMs if not neeced
+  #   and also provide the same check for all platforms
+  # the job fails when EXTENDED_TESTING is unset
+  # alternatively could upload an artifact file and abort...
+
+  check_extended_testing:
+    name: "Run EXTENDED_TESTING"
+    runs-on: ubuntu-latest
+
+    steps:
+      # --------------------------------------------------------------------------------------------
+      # cannot use secrets earlier as part of 'if:' condition
+      - name: "Runs if EXTENDED_TESTING is set"
+        env:
+          EXTENDED_TESTING: ${{ secrets.EXTENDED_TESTING }}
+          PR_TARGET_REPOSITORY: github.event.pull_request.head.repo.full_name == 'Perl/perl5' || github.repository == 'Perl/perl5'
+          CURRENT_REPOSITORY: github.repository == 'Perl/perl5'
+        run: |
+          if [ -z "${EXTENDED_TESTING}" || "${PR_TARGET_REPOSITORY}" eq 'Perl/perl5' || "${CURRENT_REPOSITORY}" eq 'Perl/perl5' ]; then
```

Comment:
Doesn't this mean that forked repos won't be able to enable extended_testing?

Reply:
this is using a `or` logic, which mean by default extended testing are not run on fork unless they set the secret variable EXTENDED_TESTING as recommended by the message